pkg/lib/bundle/build.go

```diff
@@ -0,0 +1,62 @@
+package bundle
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Create build command to build bundle manifests image
+func BuildBundleImage(directory, imageTag, imageBuilder string) (*exec.Cmd, error) {
+	var args []string
+
+	dockerfilePath := path.Join(directory, dockerFile)
+
+	switch imageBuilder {
+	case "docker", "podman":
+		args = append(args, "build", "-f", dockerfilePath, "-t", imageTag, ".")
+	case "buildah":
+		args = append(args, "bud", "--format=docker", "-f", dockerfilePath, "-t", imageTag, ".")
+	default:
+		return nil, fmt.Errorf("%s is not supported image builder", imageBuilder)
+	}
+
+	return exec.Command(imageBuilder, args...), nil
+}
+
+func ExecuteCommand(cmd *exec.Cmd) error {
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	log.Debugf("Running %#v", cmd.Args)
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("Failed to exec %#v: %v", cmd.Args, err)
+	}
+
+	return nil
+}
+
+func BuildFunc(directory, imageTag, imageBuilder, packageName, channels, channelDefault string, overwrite bool) error {
```

Comment:
I assume this is the function that the SDK will reference when it imports this package? If so, can we add a comment to this function that defines each parameter and its purpose? I know they are defined in our hidden opm command, but it would be useful to reiterate them here so that IDEs can see the doc text

Reply:
Or are they actually just importing our cobra function?

Reply:
SDK will not import our cobra code as both SDK and `opm` will use bundle library with cobra command code wrapped around it. For that reason, the cobra code for SDK and `opm` will be basically identical.

Reply:
Cool. I think this is a good idea, and I think we should start using this `lib` pattern everywhere in `opm`